Issue 22496 improve tail log

alfredo-dotcms opened the pull request:
- FE code optimization to improve log viewer performance

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

Commit:
Troubleshooting

Commit:
#22496 Improving doS() call. The function will be executed only when new lines need to be printed

Commit:
Logging more info

Commit:
Rolling back last log

alfredo-dotcms committed:
dotCMS/core#22496 log viewer optimization and code clean up

wezell commented on dotCMS/src/main/java/com/dotmarketing/servlets/taillog/TailLogServlet.java:
```diff
@@ -132,12 +129,17 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
+					response.getOutputStream().print(write);
+					response.getOutputStream().print("<script>doS('" + write + "');</script>");
+					response.getOutputStream().flush();
+				}
```
Keep in mind this is a long polling servlet - we still need it to output something and send it over the wire to keep the http request alive, even if we ignore the output on the screen.

alfredo-dotcms replied:
yes, BE code reverted

jdotcms replied:
@wezell are not going to change the approach for this?

alfredo-dotcms committed:
dotCMS/core#22496 FE code optimization to improve log viewer performance

alfredo-dotcms committed:
dotCMS/core#22496 BE reverted commit to keep alive connection

alfredo-dotcms committed:
removed HRJS lib

alfredo-dotcms committed:
FE - bug fixes on log viewer

alfredo-dotcms committed:
things tested:
- removed print 'write' variable in iframe from servlet
- FE - when new data (custom event) is listened, a loop(for) removes SCRIPT tags from the IFRAME content
- FE - removed scroll to bottom functionality
- FE - Added IF to do filtering (rewrite Black DIV) only if there's a keyword, otherwise skip it

alfredo-dotcms committed:
things tested:

- BE - add a counter to print every 60 seconds (instead of every 1 second) to keep alive the iframe connection
- BE - wrap content to be written in the Iframe with a <p> html tag to improve HTML DOM rendering
- FE - removed "logRawContent" that keeps in JS memory the whole log loaded
- FE - added back the scroll to bottom functionality
- FE - to restablish the whole log data after filtering, now we grab data from the IFrame
- FE - writing data to the filtering DIV (black DIV) is done with "insertAdjacentHTML" which improves rendering performance
- FE - doing changes to the DOM now only happens at the end of ALL processes
- FE - new content from the BE is wrapped inside a <p> html tag to improve HTML DOM rendering
- FE - filtering to highlight content changed to use "ReplaceAll"
- FE - Added input to show how many lines of code are currently on the log loaded

alfredo-dotcms committed:
things tested:
- FE - improved replaceAll to highlight keyword matches
- FE - improved cleaning LogViewer DIV
- FE - tested algorithms to highlight keyword matches and exclude rows

alfredo-dotcms committed:
FE - added conditional to skip repainting the DIV of logs, when previouly there have been no modifications to the log

alfredo-dotcms committed:
FE - fix exclude log viewer functionality not considering <BR> formatted as xHtml
